src/lib/Form/Type/Content/FieldCollectionType.php

```diff
@@ -92,6 +68,91 @@ private function isContentUpdate(array $entryOptions): bool
     {
         return !empty($entryOptions['content']) && !empty($entryOptions['contentUpdateStruct']);
     }
+
+    /**
+     * @param array<string, mixed> $entryOptions
+     */
+    private function isUserCreate(array $entryOptions): bool
+    {
+        return !empty($entryOptions['userCreateStruct']);
+    }
+
+    /**
+     * @param array<string, mixed> $entryOptions
+     */
+    private function isUserUpdate(array $entryOptions): bool
+    {
+        return !empty($entryOptions['userUpdateStruct']);
+    }
+
+    /**
+     * @param array<string, mixed> $entryOptions
+     *
+     * @return array<string, mixed> $entryOptions
+     */
+    private function dispatchFieldOptionsEvent(
+        FieldData $entryData,
+        array $entryOptions,
+        FormInterface $form
+    ): array {
+        if ($this->isContentUpdate($entryOptions)) {
```

Comment:
Can we have returns in all blocks instead of elseifs? Ideally I would split that into methods or try to at least extract some common logic to increase readability.